pkg/resource/db_cluster_parameter_group/hooks.go

```diff
@@ -173,3 +196,234 @@ func sdkTagsFromResourceTags(
 	}
 	return tags
 }
+
+func equalStrings(a, b *string) bool {
+	if a == nil {
+		return b == nil || *b == ""
+	}
+	return (*a == "" && b == nil) || *a == *b
+}
+
+// syncParameters keeps the resource's parameters in sync
+//
+// RDS does not have a DeleteParameter or DeleteParameterFromParameterGroup API
+// call. Instead, you need to call ResetDBClusterParameterGroup with a list of
+// DB Cluster Parameters that you want RDS to reset to a default value.
+func (rm *resourceManager) syncParameters(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncParameters")
+	defer func() { exit(err) }()
+
+	groupName := latest.ko.Spec.Name
+	family := latest.ko.Spec.Family
+
+	toModify, toDelete := util.ComputeParametersDelta(
+		desired.ko.Spec.ParameterOverrides, latest.ko.Spec.ParameterOverrides,
+	)
+
+	if len(toDelete) > 0 {
+		err = rm.resetParameters(ctx, family, groupName, toDelete)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(toModify) > 0 {
+		err = rm.modifyParameters(ctx, family, groupName, toModify)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// getParameters retrieves the cluster parameter group's user-defined parameters
+// (overrides) and the "statuses" of those parameter overrides.
+func (rm *resourceManager) getParameters(
+	ctx context.Context,
+	groupName *string,
+) (
+	params map[string]*string,
+	paramStatuses []*svcapitypes.Parameter,
+	err error,
+) {
+	var marker *string
+	params = make(map[string]*string)
+	for {
+		resp, err := rm.sdkapi.DescribeDBClusterParametersWithContext(
+			ctx,
+			&svcsdk.DescribeDBClusterParametersInput{
+				DBClusterParameterGroupName: groupName,
+				Source:                      aws.String(sourceUser),
```

Comment:
I have read the RDS docs but I still don't understand the significance of this. Is this relevant to our controller at all?

Reply:
@RedbackThomson there's no need to return any non-user-settable parameters, which is why this filter is passed.